security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -185,7 +185,7 @@ type SecretCache struct {
 	// close channel.
 	closing chan bool
 
-	rootCertMutex      *sync.Mutex
+	rootCertMutex      *sync.RWMutex
```

Comment:
If you use RWMutex, you need to protect all read access to the rootCert. Otherwise the code isn't consistent. Can you replace all the sc.rootCert with getRootCert()?

Reply:
Yes, sc.rootCert is only existing in the setter and getter methods.